[main] RuntimeError: set_wakeup_fd only works in main thread

Encountered during the tests on Linux (in particular, `src.tribler-core.tribler_core.components.metadata_store.tests.test_metadata_store_component.test_metadata_store_component`):

```
self = <_UnixSelectorEventLoop running=False closed=False debug=False>
sig = <Signals.SIGCHLD: 17>
callback = <bound method BaseChildWatcher._sig_chld of <asyncio.unix_events.SafeChildWatcher object at 0x7facd585bbe0>>
args = ()

    def add_signal_handler(self, sig, callback, *args):
        """Add a handler for a signal.  UNIX only.
    
        Raise ValueError if the signal number is invalid or uncatchable.
        Raise RuntimeError if there is a problem setting up the handler.
        """
        if (coroutines.iscoroutine(callback) or
                coroutines.iscoroutinefunction(callback)):
            raise TypeError("coroutines cannot be used "
                            "with add_signal_handler()")
        self._check_signal(sig)
        self._check_closed()
        try:
            # set_wakeup_fd() raises ValueError if this is not the
            # main thread.  By calling it early we ensure that an
            # event loop running in another thread cannot add a signal
            # handler.
>           signal.set_wakeup_fd(self._csock.fileno())
E           ValueError: set_wakeup_fd only works in main thread

/usr/lib/python3.8/asyncio/unix_events.py:95: ValueError

During handling of the above exception, another exception occurred:

pyfuncitem = <Function test_metadata_store_component>

    def pytest_pyfunc_call(pyfuncitem):  # type: ignore
        """
        Run coroutines in an event loop instead of a normal function call.
        """
        fast = pyfuncitem.config.getoption("--aiohttp-fast")
        if asyncio.iscoroutinefunction(pyfuncitem.function):
            existing_loop = pyfuncitem.funcargs.get('proactor_loop')\
                or pyfuncitem.funcargs.get('loop', None)
            with _runtime_warning_context():
>               with _passthrough_loop_context(existing_loop, fast=fast) as _loop:

/usr/local/lib/python3.8/dist-packages/aiohttp/pytest_plugin.py:167: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
/usr/local/lib/python3.8/dist-packages/aiohttp/pytest_plugin.py:145: in _passthrough_loop_context
    loop = setup_test_loop()
/usr/local/lib/python3.8/dist-packages/aiohttp/test_utils.py:528: in setup_test_loop
    watcher.attach_loop(loop)
/usr/lib/python3.8/asyncio/unix_events.py:927: in attach_loop
    loop.add_signal_handler(signal.SIGCHLD, self._sig_chld)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_UnixSelectorEventLoop running=False closed=False debug=False>
sig = <Signals.SIGCHLD: 17>
callback = <bound method BaseChildWatcher._sig_chld of <asyncio.unix_events.SafeChildWatcher object at 0x7facd585bbe0>>
args = ()

    def add_signal_handler(self, sig, callback, *args):
        """Add a handler for a signal.  UNIX only.
    
        Raise ValueError if the signal number is invalid or uncatchable.
        Raise RuntimeError if there is a problem setting up the handler.
        """
        if (coroutines.iscoroutine(callback) or
                coroutines.iscoroutinefunction(callback)):
            raise TypeError("coroutines cannot be used "
                            "with add_signal_handler()")
        self._check_signal(sig)
        self._check_closed()
        try:
            # set_wakeup_fd() raises ValueError if this is not the
            # main thread.  By calling it early we ensure that an
            # event loop running in another thread cannot add a signal
            # handler.
            signal.set_wakeup_fd(self._csock.fileno())
        except (ValueError, OSError) as exc:
>           raise RuntimeError(str(exc))
E           RuntimeError: set_wakeup_fd only works in main thread

/usr/lib/python3.8/asyncio/unix_events.py:97: RuntimeError
```

This was supposed to be fixed by https://github.com/Tribler/tribler/pull/6354 but it seems that it requires a bit more attention.

--- a/src/tribler-core/tribler_core/components/bandwidth_accounting/tests/test_bandwidth_accounting_component.py
+++ b/src/tribler-core/tribler_core/components/bandwidth_accounting/tests/test_bandwidth_accounting_component.py
@@ -1,5 +1,7 @@
 from unittest.mock import patch
 
+import pytest
+
 from tribler_core.components.bandwidth_accounting.bandwidth_accounting_component import BandwidthAccountingComponent
 from tribler_core.components.base import Session
 from tribler_core.components.ipv8 import Ipv8Component
@@ -10,7 +12,7 @@ from tribler_core.restapi.rest_manager import RESTManager
 
 # pylint: disable=protected-access
 
-
+@pytest.mark.asyncio
 async def test_bandwidth_accounting_component(tribler_config):
     tribler_config.ipv8.enabled = True
     components = [RESTComponent(), MasterKeyComponent(), Ipv8Component(), BandwidthAccountingComponent()]

--- a/src/tribler-core/tribler_core/components/gigachannel/tests/test_gigachannel_component.py
+++ b/src/tribler-core/tribler_core/components/gigachannel/tests/test_gigachannel_component.py
@@ -1,5 +1,7 @@
 from unittest.mock import patch
 
+import pytest
+
 from tribler_core.components.base import Session
 from tribler_core.components.gigachannel.gigachannel_component import GigaChannelComponent
 from tribler_core.components.ipv8 import Ipv8Component
@@ -11,7 +13,7 @@ from tribler_core.restapi.rest_manager import RESTManager
 
 # pylint: disable=protected-access
 
-
+@pytest.mark.asyncio
 async def test_giga_channel_component(tribler_config):
     tribler_config.ipv8.enabled = True
     tribler_config.libtorrent.enabled = True

--- a/src/tribler-core/tribler_core/components/gigachannel_manager/tests/test_gigachannel_manager_component.py
+++ b/src/tribler-core/tribler_core/components/gigachannel_manager/tests/test_gigachannel_manager_component.py
@@ -1,5 +1,7 @@
 from unittest.mock import patch
 
+import pytest
+
 from tribler_core.components.base import Session
 from tribler_core.components.gigachannel_manager.gigachannel_manager_component import GigachannelManagerComponent
 from tribler_core.components.libtorrent import LibtorrentComponent
@@ -12,6 +14,7 @@ from tribler_core.restapi.rest_manager import RESTManager
 
 # pylint: disable=protected-access
 
+@pytest.mark.asyncio
 async def test_gigachannel_manager_component(tribler_config):
     tribler_config.ipv8.enabled = True
     tribler_config.libtorrent.enabled = True

--- a/src/tribler-core/tribler_core/components/metadata_store/tests/test_metadata_store_component.py
+++ b/src/tribler-core/tribler_core/components/metadata_store/tests/test_metadata_store_component.py
@@ -1,5 +1,7 @@
 from unittest.mock import patch
 
+import pytest
+
 from tribler_core.components.base import Session
 from tribler_core.components.masterkey import MasterKeyComponent
 from tribler_core.components.metadata_store.metadata_store_component import MetadataStoreComponent
@@ -9,6 +11,7 @@ from tribler_core.restapi.rest_manager import RESTManager
 
 # pylint: disable=protected-access
 
+@pytest.mark.asyncio
 async def test_metadata_store_component(tribler_config):
     tribler_config.libtorrent.enabled = True
     tribler_config.chant.enabled = True

--- a/src/tribler-core/tribler_core/components/tests/test_base_component.py
+++ b/src/tribler-core/tribler_core/components/tests/test_base_component.py
@@ -4,6 +4,7 @@ from tribler_core.components.base import Component, ComponentError, Session, T
 
 pytestmark = pytest.mark.asyncio
 
+
 # pylint: disable=protected-access
 
 

